### PR TITLE
Search PoC: Improves initial indexing speed. Makes params configurable.

### PR DESCRIPTION
**Pull request by owensmallwood**
The indexer is currently taking a long time to build the index - around 10 mins on avg.

Running locally with the default config settings against the same database, the index builds in about 80 seconds.

**Changes**
- Only index batches once they exceed the `index_max_batch_size`, then index all remaining batches at the end.
- Index the batch for each shard concurrently, with a max concurrency of `index_workers`.
- Increase the List page size. `5000` seemed to be what worked best locally before the server started limiting the number of items in the response.
- Makes all these settings configurable, as these will need to be adjusted per environment.

Example trace for building the index. Got it down to a little over a minute, locally.
Screenshot 2024-10-25 at 4 07 26 PM

### CODE REVIEW

File: pkg/storage/unified/resource/index.go [hunk 1]
```diff
@@ -44,44 +43,86 @@ func NewIndex(s *server, opts Opts, path string, tracer tracing.Tracer) *Index {
...
-	ctx, span := i.tracer.Start(ctx, tracingPrexfixIndex+"CreateIndexBatches")
-	for _, obj := range list.Items {
-		indexableResource, err := NewIndexedResource(obj.Value)
+// IndexBatches goes through all the shards and indexes their batches if they are large enough
+func (i *Index) IndexBatches(ctx context.Context, maxSize int, tenants []string) error {
+	_, span := i.tracer.Start(ctx, tracingPrexfixIndex+"IndexBatches")
+	defer span.End()
+
+	group := errgroup.Group{}
+	group.SetLimit(500)
```
Comment (scottlepp) [inline pkg/storage/unified/resource/index.go:57]: i guess we need to store cfg on the index struct so we can set the configurable limit here

**Review by scottlepp — CHANGES_REQUESTED** [review]
looks like we need to add cfg to the index struct so we can access the settings/remove hard coded values?

File: pkg/storage/unified/resource/index.go (hunk 1 above)
Comment (owensmallwood) [reply pkg/storage/unified/resource/index.go:57]: ah, good catch! The config settings are on Index.Opts already, so updated this to use that 👍

**Review by scottlepp — APPROVED** [review]
this is faster.  though for indexing 100k items it went from 2.4 seconds to 2 seconds.  so not a huge change.  seems something else is happening to cause the delays we saw in the tracing/logs

**Pull request merged**